README.md

```diff
@@ -1,138 +1,221 @@
+
+Below these keys are any number of arbitrarily named dependency lists (i.e. `build`, `test`, `libcuml_build`, `cuml_build`, etc.).
+
+An example of the above structure is exemplified below:
+
+```yaml
 dependencies:
-  - cudatoolkit==11.6.0
-  - django=3
-  - gcc_linux-64=9.*
-  - nvcc_linux-64=11.6
-  - pytorch
-  - sysroot_linux-64==2.17
-  - pip:
-      - random_dep
-      - snakes
-name: build_cuda-11.6_arch-x86_64
+  conda_requirements: # common dependencies between `conda` environment & requirements.txt files
+    common: # common between archs/cudas
+      build: # arbitrarily named dependency list
+        - common_build_dep
+      test: # arbitrarily named dependency list
+        - pytest
+    x86_64-11.5: # common dependencies specific to x86_64-11.5
+      build:
+        - a_random_x86_115_specific_dep
+  conda: # dependencies specific to conda environment files
+    common:
+      build:
+        - cupy
+        - pip: # supports `pip` key for `conda` environment files
+            - some_random_dep
+    x86_64-11.5:
+      build:
+        - cudatoolkit=11.5
+    x86_64-11.6:
+      build:
+        - cudatoolkit=11.6
+  requirements: # dependencies specific to requirements.txt files
+    x86_64-11.5:
+      build:
+        - another_random_dep=11.5.0
+    x86_64-11.6:
+      build:
+        - another_random_dep=11.6.0
 ```
 
-Mutliple environments can be specified. For instance, the following `envs.yaml` file:
+## How Dependency Lists Are Merged
+
+The information from the top-level `files` and `dependencies` keys are used to determine which dependencies should be included in the final output of the generated dependency files.
+
+Consider the following top-level `files` key configuration:
 
 ```yaml
-envs:
-  build: # `build` environment configuration
-    matrix:
-      cuda_version: ["11.5"]
-      arch: [x86_64]
-    includes:
-      - build
-  build_and_test: # `build_and_test` environment configuration
+files:
+  all:
+    generate: conda
+    conda_dir: conda/environments
+    requirements_dir: python/cudf
     matrix:
-      cuda_version: ["11.5"]
-      arch: [x86_64]
+      cuda_version: ["11.5", "11.6"]
+      arch: [x86_64, arm]
     includes:
       - build
       - test
-channels:
-  - rapidsai
-  - nvidia
-specifics:
-build:
-  - django=3
-  - pytorch
-  - pip:
-      - snakes
-test:
-  - pytest
 ```
 
-will create `build_and_test_cuda-11.5_arch-x86_64.yaml` and `build_cuda-11.5_arch-x86_64.yaml`, with the only difference being that the `build_and_test_cuda-11.5_arch-x86_64.yaml` file will also include `pytest` as a dependency.
+For the `11.5` and `x86_64` matrix combination, the following dependency lists would be merged if they exist:
+
+- `conda_requirements.common.build`
+- `conda_requirements.x86_64-11.5.build`
+- `conda.common.build`
+- `conda.x86_64-11.5.build`
+- `conda_requirements.common.test`
+- `conda_requirements.x86_64-11.5.test`
+- `conda.common.test`
+- `conda.x86_64-11.5.test`
```

Comment:
Sort these so that the lines are more clearly grouped:
```suggestion
- `conda_requirements.common.build`
- `conda_requirements.common.test`
- `conda_requirements.x86_64-11.5.build`
- `conda_requirements.x86_64-11.5.test`
- `conda.common.build`
- `conda.common.test`
- `conda.x86_64-11.5.build`
- `conda.x86_64-11.5.test`
```